### PR TITLE
(introspection) - Implement initial @urql/introspection package

### PR DESCRIPTION
**Pull request by kitten**
Related #974

The new `@urql/introspection` utilities aim to contain all kinds of functions that may be used to treat client-side GraphQL schemas, currently: to minify them. This may be used to minify an `IntrospectionQuery` to get it down to a minimum possible size.

This has been tested against the GitHub GraphQL API schema and it brings it down from `49.4kB` minzipped to `29.6kB` minzipped. The uncompressed sizes are `1.1MB` and `551KB`.

We've determined that it's _not_ effective to translate the GraphQL Introspection Query to a schema string and turning that back to a `GraphQLSchema`. The reduction is about a third of size, but this does pull in about 15-20KB more from `graphql`. Since we're dealing with a worst-case scenario sized schema here, this doesn't seem worth it, since that extra code increases the parsing time much more (including the GraphQL schema parsing time) than pure JSON.

### CODE REVIEW

File: packages/introspection/src/minifyIntrospectionQuery.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        `Unrecognized type reference of type: ${(fromType as any).kind}.`
+      );
+  }
+};
+
+const minifyIntrospectionType = (
+  type: IntrospectionType
+): IntrospectionType => {
+  switch (type.kind) {
+    case 'OBJECT': {
```
Comment (JoviDeCroock) [inline packages/introspection/src/minifyIntrospectionQuery.ts:48]: Can we maybe use the `Kind` exported from graphql for safety?

**Review by JoviDeCroock — APPROVED** [review]
Nit, not blocking. Awesome work Phil

File: packages/introspection/src/minifyIntrospectionQuery.ts (hunk 1 above)
Comment (kitten) [reply packages/introspection/src/minifyIntrospectionQuery.ts:48]: I'm not sure if it's the same 🤔

Comment (JoviDeCroock) [reply packages/introspection/src/minifyIntrospectionQuery.ts:48]: Oh all right! thanks for checking 👍

**Comment by frederikhors** [comment]
Amazing! Now it would be nice to understand how I use it! LOL! :smile:

**Pull request merged**